code/game/objects/items/devices/gps.dm

```diff
@@ -18,11 +18,17 @@ GLOBAL_LIST_EMPTY(GPS_list)
 	. += span_notice("Alt-click to switch it [tracking ? "off":"on"].")
 
 /obj/item/gps/Initialize()
-	. = ..()
-	GLOB.GPS_list += src
-	name = "global positioning system ([gpstag])"
-	if(tracking) //Some roundstart GPS are off.
-		add_overlay("working")
+    . = ..()
+    GLOB.GPS_list += src
+    name = "global positioning system ([gpstag])"
+    if(tracking) //Some roundstart GPS are off.
+        add_overlay("working")
+    addtimer(CALLBACK(src, .proc/InitMob), 2 SECONDS)
+
+/obj/item/gps/proc/InitMob()
+    var/mob/living/carbon/human/potential = src.loc.loc
```

Comment:
what the fuck

Reply:
jamie's hacky attempt to make the GPS name itself to whoever it spawns on